### ISSUE TITLE
can`t use msys2 unix utilities to build tbb

### ISSUE THREAD

**Issue by dr-pain** [intel/tbb#217]
It failed with message:
'\"cscript /nologo /E:jscript ./build/detect.js  /arch \""' is not recognized as an internal or external command, operable program or batch file.
'\"cscript /nologo /E:jscript ./build/detect.js  /runtime \""' is not recognized as an internal or external command, operable program or batch file.
build/common.inc:81: *** Architecture not detected.  Stop.

Could you
- update "build directions" section of doc and list unix utilities sets which were tested
or
- mentioned msys2 as "not supported"
or
- fix msys2 build issues?

**Comment by ivankochin** [comment]
Hi, @dr-pain 
Thanks for reporting!
msys2 environment is not currently supported.

**Comment by dr-pain** [comment]
@ivankochin I know it. Did you read "could you" section in my post?

**Comment by alexey-katranov** [comment]
We will consider if it is feasible to support msys2. 

> ... list unix utilities sets which were tested

The list of officially supported/tested OSes, compilers and tools (including xcode and glibc versions): https://github.com/oneapi-src/oneTBB/blob/tbb_2020/doc/Release_Notes.txt#L60-L97. Do you need some additional system info?

**Comment by dr-pain** [comment]
> Do you need some additional system info?

yes. in build/index.html I see "On Windows OS, if a UNIX* emulator is used to run GNU make, it should be able to run Windows OS utilities and commands.". But I can`t find list of supported UNIX* emulators in system reqs.
it will be ok if you publish such list (or at least one recommended emulator) somewhere in TBB docs.
this issue isn`t direct request to support msys.

**Comment by alexey-katranov** [comment]
Thank you for the clarification. I agree that the documentation does not state it explicitly, "UNIX* emulator" is supported by the community, i.e. there is no official support, so nothing is stated in the official list of supported platforms. MinGW is supported de facto (it is mentioned in https://github.com/oneapi-src/oneTBB/blob/tbb_2020/build/index.html#L133).

**Comment by dr-pain** [comment]
> it is mentioned in https://github.com/oneapi-src/oneTBB/blob/tbb_2020/build/index.html#L133
it mentioned in context of gcc compiler usage at windows
I`m asking to put MinGW mention to "Software prerequisites".

**Comment by saudet** [comment]
I've created a patch to make it work with MSYS2 here:
https://github.com/bytedeco/javacpp-presets/blob/ci/dnnl/tbb-windows.patch
It probably still works with MinGW/MSYS 1.x, but I haven't tested it, and being abandonware, probably not worth worrying over.
@ivankochin @alexey-katranov If you would like to have this integrated into this repository, let me know and I'll open a PR!

**Comment by anton-potapov** [comment]
@dr-pain , [oneTBB v2021.4](https://github.com/oneapi-src/oneTBB/releases/tag/v2021.4.0)  should work with msys2 (after [351](https://github.com/oneapi-src/oneTBB/pull/351) landed)

**Comment by anton-potapov** [comment]
#675 is to mention MinGW in the known to work systems
